agent/tool-scripts/README.md

```diff
@@ -0,0 +1,219 @@
+# Tool Scripts (`agent/tool-scripts`)
+
+The "scripts" in this directory represent the "tools" that users can register
+to collect data from the systems related to a workload.  Each tool available
+for use with a `pbench-register-tool --name=<tool>` CLI invocation is listed
+in the `meta.json` file as either "transient" or "persistent".
+
+The Tool Meister running on each host where tools are registered is
+responsible for invoking that tool script to collect its generated data.  The
+commands to start and stop tools are sent to each Tool Meister at the
+appropriate time via the `pbench-init/start/stop/end/send/postprocess-tools`
+CLI interfaces.
+
+The tool scripts in this directory provide the interfaces used by the Tool
+Meister for checking for prerequisites, starting the tool data capture
+("datalog" scripts), stopping the tool data capture, and optionally
+post-processing the captured tool data ("postprocess" scripts).
+
+Most of the "tools" are symlinks to one script named `base-tool`.  This allows
+us to share all the common behaviors between the tools, while hanlding
+differences where appropriate.
+
+The `external-data-source`, `node-exporter`, and `dcgm` "tools" are handled
+separately since they operate a bit different from the rest.  See below for
+more details.
+
+## The Tool Life Cycle
+
+There are 4 phases, if you will, that the Tool Meister sub-system offers. The
+first phase is for executing all the "persistent" tools.  Once the Tool
+Meister sub-system is up and running, this phase is started first (required),
```

Comment:
Does the user have any say? It's not clear to me what "required" means here.

Reply:
I guess they do if they write their own bench scripts.

Reply:
Right, we offer `pbench-start-tools` and friends as part of the CLI interface we support, for better or worse.